torch/lib/THD/base/channels/DataChannelTCP.cpp

```diff
@@ -196,70 +196,99 @@ DataChannelTCP::~DataChannelTCP()
 
 
 void DataChannelTCP::listen(std::uint16_t port = 0) {
-  SYSCHECK(_socket = ::socket(PF_INET, SOCK_STREAM, 0))
+  struct addrinfo hints, *res = NULL;
+
+  memset(&hints, 0x00, sizeof(hints));
+  hints.ai_flags = AI_PASSIVE;
+  hints.ai_family = AF_UNSPEC; // either IPv4 or IPv6
+  hints.ai_socktype = SOCK_STREAM; // TCP
+
+  // `getaddrinfo` will sort addresses according to RFC 3484 and can be tweeked
```

Comment:
I think we can just leave it as is, maybe it will work out fine. I just know that I wouldn't be happy having to tell users that they should change a system-wide config file to make it work 😉